docs/_ext/exampleinclude.py

```diff
@@ -0,0 +1,51 @@
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.directives.code import LiteralIncludeReader
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util.nodes import set_source_info
+
+
+class ExampleInclude(SphinxDirective):
+    """Extract the text between two tag"""
+
+    has_content = True
+
+    git_repo = "https://github.com/astronomer/astronomer-providers/tree/main/"
+
+    option_spec = {
+        "dedent": int,
+        "linenos": directives.flag,
+        "lineno-start": int,
+        "lineno-match": directives.flag,
+        "language": directives.unchanged_required,
+        "pyobject": directives.unchanged_required,
+        "lines": directives.unchanged_required,
+        "start-after": directives.unchanged_required,
+        "end-before": directives.unchanged_required,
+        "class": directives.class_option,
+        "name": directives.unchanged,
+    }
+
+    def run(self):
+        """Extract text and return list of node"""
+        location = self.state_machine.get_source_and_line(self.lineno)
+        rel_filename, filename = self.env.relfn2path(self.content[0])
+        reader = LiteralIncludeReader(filename, self.options, self.config)
+        text, lines = reader.read(location=location)
+        example_git_location = f"# {self.git_repo}{rel_filename[3:]}"
+        text = text.__add__(example_git_location)
```

Comment:
This is currently coming as text but I want to make it a clickable link like HTML href tag.  
@kaxil any thought on it how I can achieve it?

<img width="679" alt="Screenshot 2022-06-27 at 7 55 35 PM" src="https://user-images.githubusercontent.com/98807258/175964695-8361e3b1-3a9d-4e78-a4a1-172b6988f618.png">

Reply:
in Airflow we use https://www.sphinx-doc.org/en/master/usage/extensions/viewcode.html -> https://github.com/apache/airflow/blob/main/docs/exts/exampleinclude.py